test/unit/specs/hive/key-value.js

```diff
@@ -0,0 +1,254 @@
+        await expect(() => store.set('k', 'v', { expiration: false })).to.throw(errorMsg2)
+        await expect(() => store.set('k', 'v', { expiration: true })).to.throw(errorMsg2)
+        await expect(() => store.set('k', 'v', { expiration: '' })).to.throw(errorMsg2)
+        await expect(() => store.set('k', 'v', { expiration: 'foo' })).to.throw(errorMsg2)
+        await expect(() => store.set('k', 'v', { expiration: NaN })).to.throw(errorMsg2)
+        await expect(() => store.set('k', 'v', { expiration: () => undefined })).to.throw(errorMsg2)
+
+        const errorMsg3 = 'Condition must be one of this values: IfExists, IfNotExists, Always.'
+
+        await expect(() => store.set('k', 'v', { condition: null })).to.throw(errorMsg3)
+        await expect(() => store.set('k', 'v', { condition: false })).to.throw(errorMsg3)
+        await expect(() => store.set('k', 'v', { condition: true })).to.throw(errorMsg3)
+        await expect(() => store.set('k', 'v', { condition: '' })).to.throw(errorMsg3)
+        await expect(() => store.set('k', 'v', { condition: 'foo' })).to.throw(errorMsg3)
+        await expect(() => store.set('k', 'v', { condition: NaN })).to.throw(errorMsg3)
+        await expect(() => store.set('k', 'v', { condition: () => undefined })).to.throw(errorMsg3)
+      })
+    })
+
+    describe('Increment', () => {
+      it('success', async () => {
+        store = Backendless.Hive(hiveName).KeyValueStore('storeKey')
+
+        const req1 = prepareMockRequest(fakeResult)
+
+        const result1 = await store.increment(10)
+
+        expect(req1).to.deep.include({
+          method: 'PUT',
+          path  : `${storeUrl}/storeKey/increment?value=10`,
+        })
+
+        expect(result1).to.be.eql(fakeResult)
+      })
+
+      it('fails when store created without key', async () => {
+        const errorMsg = 'Store must be created with store key.'
+
+        await expect(() => store.increment(10)).to.throw(errorMsg)
+      })
+
+      it('fails when value is invalid', async () => {
+        store = Backendless.Hive(hiveName).KeyValueStore('storeKey')
+
+        const errorMsg = 'Value must be provided and must be a number.'
+
+        await expect(() => store.increment(undefined)).to.throw(errorMsg)
+        await expect(() => store.increment(null)).to.throw(errorMsg)
+        await expect(() => store.increment(false)).to.throw(errorMsg)
+        await expect(() => store.increment('')).to.throw(errorMsg)
+        await expect(() => store.increment('123')).to.throw(errorMsg)
+        await expect(() => store.increment(NaN)).to.throw(errorMsg)
+        await expect(() => store.increment(true)).to.throw(errorMsg)
+        await expect(() => store.increment(() => undefined)).to.throw(errorMsg)
+        await expect(() => store.increment({})).to.throw(errorMsg)
+      })
+    })
+
+    describe('Decrement', () => {
+      it('success', async () => {
+        store = Backendless.Hive(hiveName).KeyValueStore('storeKey')
```

Comment:
doesn't the `store` exist yet?

Reply:
exist, but this method requires a store key name.

So I don't think it's good for now to move it into `beforeEach`, because we use it only 4 times in different places.

If at future, something changed, we will do it